### ISSUE TITLE
URL blank encode problem

### ISSUE THREAD

**Issue by wanggp3** [aio-libs/aiohttp#808]
```
#testfile.py
import asyncio,aiohttp

url  = 'http://www.gilt.com/search?q.query=Sam Edelman&q.rows=48&q.category=womens-coats&q.start=-48'

@asyncio.coroutine
def fetch():
    session = aiohttp.ClientSession()#cookies={'currencyid':'2','topcatid':'9057'})
    r = yield from session.get(url)
    body = yield from r.read()
    #print(body)
    print('aiohttp statuss = %s ' % r.status)
    session.close()


def get():
    import requests
    r = requests.get(url)
    print('requests status = %s' % r.status_code)

get()
print('*' * 10)

loop = asyncio.get_event_loop()
loop.run_until_complete(fetch())
loop.close()
```

In the [RFC1738](https://tools.ietf.org/html/rfc1738)

>    Unsafe:
>    Characters can be unsafe for a number of reasons.  The space
>    character is unsafe because significant spaces may disappear and
>    insignificant spaces may be introduced when URLs are transcribed or
>    typeset or subjected to the treatment of word-processing programs.
>    The characters "<" and ">" are unsafe because they are used as the
>    delimiters around URLs in free text; the quote mark (""") is used to
>    delimit URLs in some systems.  The character "#" is unsafe and should
>    always be encoded because it is used in World Wide Web and in other
>    systems to delimit a URL from a fragment/anchor identifier that might
>    follow it.  The character "%" is unsafe because it is used for
>    encodings of other characters.  Other characters are unsafe because
>    gateways and other transport agents are known to sometimes modify
>    such characters. These characters are "{", "}", "|", "\", "^", "~",
>    "[", "]", and "`".
>    All unsafe characters must always be encoded within a URL. For
>    example, the character "#" must be encoded within URLs even in
>    systems that do not normally deal with fragment or anchor
>    identifiers, so that if the URL is copied into another system that
>    does use them, it will not be necessary to change the URL encoding.

If we pass a url with blank, the aiohttp will fail to get it, but the requests can. So, whether the aiohttp  should encode the url with blank first?

**Comment by asvetlov** [comment]
Agree. Would you provide a Pull Request?

**Comment by wanggp3** [comment]
Yes, I have provided it

**Comment by asvetlov** [comment]
Fixed by #1217
